marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -4501,6 +4506,18 @@ private <U extends OkHttpResultIterator> U makeResults(
       getEntity(body, MimeMultipart.class) : null;
 
     List<BodyPart> partList = getPartList(entity);
+      try {
+          Headers trailer = response.trailers();
+          String code = trailer.get("ml-error-code");
+          String msg = trailer.get("ml-error-message");
+          String sha = trailer.get("ml-content-sha256");
+
+          if (code != null && !"N/A".equals(code)) {
+              throw new RuntimeException("code = " + code + ", msg = " + msg + ", sha256 = " + sha);
```

Comment:
Is it possible to map to one of the MarkLogic Java exceptions in the same cases that would have this mapping without a chunked response?

That is, is it possible to refactor the existing exception construction code to be callable with either an HTTP error code and message or a trailer error code and message?